bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h

```diff
@@ -0,0 +1,167 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/**
+ * @file
+ * @brief ESP BSP: ESP32-S3-Korvo-1
+ */
+
+#pragma once
+
+#include "iot_button.h"
+#include "led_indicator.h"
+
+/**************************************************************************************************
+ *  ESP32-S3-Korvo-1 pinout
+ **************************************************************************************************/
+
+/* Leds */
+#define BSP_LED_RGB_GPIO        (GPIO_NUM_19)
+#define BSP_LED_NUM             (12)
+
+/* Buttons */
+#define BSP_BUTTONS_IO          (GPIO_NUM_8) // All 6 buttons mapped to this GPIO
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Default LED effects */
+enum {
+    BSP_LED_ON,
+    BSP_LED_OFF,
+    BSP_LED_BLINK_FAST,
+    BSP_LED_BLINK_SLOW,
+    BSP_LED_BREATHE_FAST,
+    BSP_LED_BREATHE_SLOW,
+    BSP_LED_MAX,
+};
+
+/**************************************************************************************************
+ *
+ * Buttons interface
+ *
+ * Example configuration:
+ * \code{.c}
+ * button_handle_t btns[BSP_BUTTON_NUM];
+ * bsp_iot_button_create(btns, NULL, BSP_BUTTON_NUM);
+ * iot_button_register_cb(btns[0], ...
+ * \endcode
+ **************************************************************************************************/
+typedef enum {
+    BSP_BUTTON_REC = 0,
+    BSP_BUTTON_MODE,
+    BSP_BUTTON_PLAY,
+    BSP_BUTTON_SET,
+    BSP_BUTTON_VOLDOWN,
+    BSP_BUTTON_VOLUP,
+    BSP_BUTTON_NUM,
+} bsp_button_t;
+
+/**************************************************************************************************
+ *
+ * ADC interface
+ *
+ * There are multiple devices connected to ADC peripheral:
+ *  - Buttons
+ *
+ * After initialization of ADC, use adc_handle when using ADC driver.
+ **************************************************************************************************/
+
+#define BSP_ADC_UNIT     ADC_UNIT_1
+
+/**
+ * @brief Initialize ADC
+ *
+ * The ADC can be initialized inside BSP, when needed.
+ *
+ * @param[out] adc_handle Returned ADC handle
+ */
+esp_err_t bsp_adc_initialize(void);
+
+
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
+/**
+ * @brief Get ADC handle
+ *
+ * @note This function is available only in IDF5 and higher
+ *
+ * @return ADC handle
+ */
+adc_oneshot_unit_handle_t bsp_adc_get_handle(void);
+#endif
+
+/**
+ * @brief Initialize all buttons
+ *
+ * Returned button handlers must be used with espressif/button component API
+ *
+ * @param[out] btn_array      Output button array
+ * @param[out] btn_cnt        Number of button handlers saved to btn_array, can be NULL
+ * @param[in]  btn_array_size Size of output button array. Must be at least BSP_BUTTON_NUM
+ * @return
+ *     - ESP_OK               All buttons initialized
+ *     - ESP_ERR_INVALID_ARG  btn_array is too small or NULL
+ *     - ESP_FAIL             Underlaying iot_button_create failed
+ */
+esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int btn_array_size);
+
+/**
```

Comment:
Please, can you separate button section and LED section with this (it is better for reading then, if each section has this divider):
```
/**************************************************************************************************
 *
 * Buttons
 *
 **************************************************************************************************/
```
and

```
/**************************************************************************************************
 *
 * LEDs
 *
 **************************************************************************************************/
```